### PR TITLE
protect filesystems from signals

### PR DESCRIPTION
**Pull request by paulcallen**
Signed-off-by: Paul Allen <paul.c.allen@microsoft.com>

### CODE REVIEW

File: kernel/lockfs.c [hunk 1]
```diff
@@ -19,42 +20,53 @@ typedef struct lockfs
...
-    myst_mutex_unlock(sig_handler->lock);
+#if ULONG_MAX == 0xffffffff
+    set->__bits[0] = 0x7ffffffful;
+    set->__bits[1] = 0xfffffffcul;
+    if (_NSIG > 65)
+    {
+        set->__bits[2] = 0xfffffffful;
+        set->__bits[3] = 0xfffffffful;
+    }
+#else
```
Comment (mikbras) [inline kernel/lockfs.c:49]: ulong max is always 64-bit on x86_64.

**Review by mikbras — CHANGES_REQUESTED** [review]
I created a new branch (paulcallen/lockfs-signal-v2) with two minor modifications:
- included musl sigfillset.c in kernel directly to avoid licensing issue from cut-n-paste.
- factored out LOCK/UNLOCK macros by moving conditional check inside lock() function.

**Comment by mikbras** [comment]
I can't see anything functionally wrong but needed to at least work around licensing implication of including musl code in the kernel.

**Comment by paulcallen** [comment]
pulled in the sigfillset.c change from your branch

**Comment by paulcallen** [comment]
unfortunately, moving the signal disabling into the lock creates a race condition that can leave the lock active when an exception occurs.

**Review by mikbras — APPROVED** [review]
Thank you for doing this. Looks good.

**Pull request merged**